src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -421,6 +417,13 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
                                                         temporaryTableReference.getTableId(), table).execute();
       }
 
+      if (operation.equals(Operation.INSERT)) {
+        handleInsertOperation(tableRef, writeDisposition, loadConfig.getDestinationEncryptionConfiguration(), projectId,
+                              jobId, dataset, tableExists);
+      } else {
+        operationAction(tableRef, kmsKeyName, getJobIdForUpdateUpsert(conf), conf);
```

Comment:
nitpick: Since this now covers the UPDATE/UPSERT operation, maybe this method should be renamed to handleUpdateUpsertOperation?

Reply:
renamed it.